docs/en/thrift.md

```diff
@@ -0,0 +1,140 @@
+- Support synchronous, asynchronous, semi-synchronous accesses etc. Support [ParallelChannel etc](combo_channel.md) to define access patterns declaratively.
+- Support various connection types(short, connection pool). Support timeout, backup request, cancellation, tracing, built-in services, and other benefits offered by brpc.
+
+# Compile and Run
+In order to not depend on and compile with thrift library for most of the users, the thrift protocol wasn't supported in brpc by default. configure brpc with --with-thrift if you want to enable thrift protocol in brpc(actually it works with a Macro ENABLE_THRIFT_FRAMED_PROTOCOL)
+
+Install Thrift in Ubuntu
+```bash
+wget http://www.us.apache.org/dist/thrift/0.9.3/thrift-0.9.3.tar.gz
+tar -xf thrift-0.9.3.tar.gz
+cd thrift-0.9.3/
+./configure --prefix=/usr --with-ruby=no --with-python=no --with-java=no --with-go=no --with-perl=no --with-php=no --with-csharp=no --with-erlang=no --with-lua=no --with-nodejs=no
+make -j 3 -s
+sudo make install
+```
+Configure with thrift support
+```bash
+sh config_brpc.sh --headers=/usr/include --libs=/usr/lib --nodebugsymbols --with-thrift
+```
+Link with libbrpc_thrift.a if the user want to enable thrift protocol
+
+# Thrift message definition, echo.thrift:
+```c++
+namespace cpp example
+
+struct EchoRequest {
+    1: required string data;
+    2: required i32 s;
+}
+
+struct EchoResponse {
+    1: required string data;
+}
+
+service EchoService {
+    EchoResponse Echo(1:EchoRequest request);
+}
+```
+
+
+# Client asscess thrift server
+
+create a Channel to access thrift server:
+
+```c++
+#include <brpc/channel.h>
+#include <brpc/details/thrift_utils.h>
+#include <brpc/thrift_message.h>
+...
+
+DEFINE_string(server, "0.0.0.0:8019", "IP Address of thrift server");
+DEFINE_string(load_balancer, "", "The algorithm for load balancing");
+...
+  
+brpc::ChannelOptions options;
+options.protocol = brpc::PROTOCOL_THRIFT;
+brpc::Channel thrift_channel;
+if (thrift_channel.Init(Flags_server.c_str(), FLAGS_load_balancer.c_str(), &options) != 0) {
+   LOG(ERROR) << "Fail to initialize thrift channel";
+   return -1;
+}
+...
+```
+construct thrift request and send to server, ThriftTemplateMessage is a template class, the native thrift message was delegated by and can be accessed directly by raw() method.
+
+```c++
+ // wrapper thrift raw request into ThriftMessage
+ // example::[EchoRequest/EchoResponse]is thrift native message generated by thrift toolkits
+ brpc::ThriftTemplateMessage<example::EchoRequest> req;
+ brpc::ThriftTemplateMessage<example::EchoResponse> res;
+
+ req.raw().data = "hello";
+
+ cntl.set_thrift_method_name("Echo");
+
+ channel.CallMethod(NULL, &cntl, &req, &res, NULL);
+
+ if (cntl.Failed()) {
+     LOG(ERROR) << "Fail to send thrift request, " << cntl.ErrorText();
+     return -1;
+ } 
+```
+
+# Server process with thrift message
+User need to implement it's own thrift handler which was inherited from brpc::ThriftService

```

Comment:
User ->Users
it's -> their
brpc::ThriftService -> brpc::ThriftService.

Reply:
done